### PR TITLE
Get everything working

### PR DESCRIPTION
**Pull request by CruzBishop**
image

Well, that was fun 😬 

 * Did some refactoring to make things a bit easier
 * Added some colors, not sure what was best so just went for contrast
 * Added a gradle run config
 * Band-aided tile loading, will probably have to be changed in the future. Maybe like how the colors are?
 * Fixed sprite rendering

Update: Dynamically get screen width and height

**It seems to go a bit weird if you resize the window. Not sure why, I need more coffee, sorry**

Just noticed I don't think goals are loading, will have to look at that later

@HoldYourWaffle Have fun, mate 😄

### CODE REVIEW

**Comment by CruzBishop** [comment]
@HoldYourWaffle Ok, I'm done for now. Ended up disabling window resizing so it doesn't go all weird for now 👍

**Comment by HoldYourWaffle** [comment]
Wow you went alot deeper & further than I expected :O

I'm kinda just dying on the couch here at 5 in the morning so I can't really take a look at the moment (mobile github hates me for some reason).

I can see that you added a 'rub configuration' to gradle, not sure what's that supposed to do? (Or it's reallt obvious and I just can't think of it because it feels like my tummy is trying to process a rock ☺)

**Comment by CruzBishop** [comment]
Hey mate, at work so have to be short, it adds ```gradle run``` so you can
well, run it 😛 see you later, mate 😁

On 31 Mar. 2017 1:26 pm, "Ravi van Rooijen" <notifications@github.com>
wrote:

Wow you went alot deeper & further than I expected :O

I'm kinda just dying on the couch here at 5 in the morning so I can't
really take a look at the moment (mobile github hates me for some reason).

I can see that you added a 'rub configuration' to gradle, not sure what's
that supposed to do? (Or it's reallt obvious and I just can't think of it
because it feels like my tummy is trying to process a rock ☺)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/HoldYourWaffle/Happy-Traffic/pull/3#issuecomment-290606027>,
or mute the thread

**Comment by HoldYourWaffle** [comment]
But what's the point of that if you can also just run it from intelliji/eclipse/other-ide....
Is it just to be nice to people that don't use an IDE for some reason?

PS Installing linux on my ooold and sloooow laptop now so I can look & code from the comfort of my bed 😎 
PPS You may need to fix your email replies

**Comment by HoldYourWaffle** [comment]
Well that was fun. Now that linux is working and gradle likes my JAVA_HOME ( :watch: ), I can finally take a look at this.

**Comment by CruzBishop** [comment]
Just kidding 😛

**Comment by CruzBishop** [comment]
Error: java installation not found

On 31 Mar. 2017 9:46 pm, "Ravi van Rooijen" <notifications@github.com>
wrote:

Well that was fun. Now that linux is working and gradle likes my JAVA_HOME
( ⌚️ ), I can finally take a look at this.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/HoldYourWaffle/Happy-Traffic/pull/3#issuecomment-290691274>,
or mute the thread

**Comment by CruzBishop** [comment]
I see what you mean about fixing my email replies @HoldYourWaffle - Oops 😞

**Comment by HoldYourWaffle** [comment]
Your joke didn't really come through because your "just kidding" came before the error.... (You confused the heck out of me)

Anyway now that I've eaten lunch and eclipse likes my run settings ( :watch:  again :cry: ) I can **really** start looking in to this :)

File: src/main/java/info/zthings/ht/test/ExampleGame.java [hunk 1]
```diff
@@ -19,52 +19,64 @@
...
 public class ExampleGame extends ApplicationAdapter {
 	public Tile[][] map;
-	public RenderContextProvider g;
+	public int height;
+	public int width;
+	public RenderContextProvider contextProvider;
 	
 	@Override
 	public void create() {
-		g = new RenderContextProvider(new SpriteBatch(), new ShapeRenderer());
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: Wow you already refactored all my "short-dev-private-code" variable names :open_mouth: 

<small>you may have already noticed, but I published this code kindof prematurely just so I could ask you if I'm not going insane... so I'm sorry you had to handle my dirt code :-1: </small>

File: src/main/java/info/zthings/ht/test/ExampleGame.java [hunk 2]
```diff
@@ -19,52 +19,64 @@
...
 		System.out.println("Map is ["+map.length+"x"+map[0].length+"]");
 		
 		HashMap<String, Class<? extends Tile>> tiletypes = new HashMap<>();
-		Reflections ref = new Reflections("info.zthings.ht");
-		for (Class<? extends Tile> c : ref.getSubTypesOf(Tile.class)) tiletypes.put(c.getSimpleName().toLowerCase(), c);
+		Reflections reflections = new Reflections("info.zthings.ht");
+		for (Class<? extends Tile> reflectionClass : reflections.getSubTypesOf(Tile.class)) tiletypes.put(reflectionClass.getSimpleName().toLowerCase(), reflectionClass);
 
-		for (String k : json.keySet()) {
-			if (k == "width" || k == "height" || k == "version") continue;
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: Not sure what I was thinking here, altough it did work...

File: src/main/java/info/zthings/ht/test/ExampleGame.java [hunk 3]
```diff
@@ -73,21 +85,27 @@ public void create() {
 	public void render() {
 		Gdx.gl.glClearColor(1, 0, 0, 1);
 		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
-		
-		int ssize = 100; //TODO should be in ui
+
+        int tileWidth = Gdx.graphics.getBackBufferWidth() / width;
+        int tileHeight = Gdx.graphics.getBackBufferHeight() / height;
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: Daejm you even fixed my nasty hardcoded size constant :+1:

File: src/main/java/info/zthings/ht/test/ExampleGame.java [hunk 4]
```diff
@@ -73,21 +85,27 @@ public void create() {
...
 		
 		for (int y=0; y<map.length; y++) for (int x=0; x<map[y].length; x++) {
-			map[x][y].debugRender(g, x*ssize, y*ssize, ssize);
+		    if (map[x][y] == null) {
+		        //Nothing at this location
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: And yet again Mr. @CruzBishop is fixing my stupid-first-try-dirt-code before I even run into it :1st_place_medal:

File: src/main/java/info/zthings/ht/test/ExampleGame.java [hunk 5]
```diff
@@ -73,21 +85,27 @@ public void create() {
...
+            }
+			map[x][y].debugRender(contextProvider, x*tileWidth, y*tileHeight, tileWidth, tileHeight);
 			
-			g.sr.begin(ShapeType.Filled);
-			g.sr.setColor(map[x][y].getDebugCol());
-			g.sr.rect(x*ssize, y*ssize, ssize, ssize);
+			contextProvider.spriteRenderer.begin(ShapeType.Filled);
+			contextProvider.spriteRenderer.setColor(map[x][y].getDebugCol());
+			contextProvider.spriteRenderer.rect(x*tileWidth, y*tileHeight, tileWidth, tileHeight);
+			contextProvider.spriteRenderer.end();
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: :open_mouth: :1st_place_medal: :100:

File: src/main/java/info/zthings/ht/test/RenderContextProvider.java [hunk 6]
```diff
@@ -5,18 +5,18 @@
...
 public class RenderContextProvider implements Disposable {
-	public SpriteBatch sb;
-	public ShapeRenderer sr;
+	public SpriteBatch spriteBatch;
+	public ShapeRenderer spriteRenderer;
 	
-	public RenderContextProvider(SpriteBatch sb, ShapeRenderer sr) {
-		this.sb = sb;
-		this.sr = sr;
+	public RenderContextProvider(SpriteBatch spriteBatch, ShapeRenderer sr) {
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/RenderContextProvider.java]: So why did you rename sb -> spriteBatch but not sr -> shapeRenderer?

File: src/main/java/info/zthings/ht/test/ExampleGame.java (hunk 1 above)
Comment (CruzBishop) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: All good, mate 👍  I was going to leave it but thought I might as well change it as I was debugging

File: src/main/java/info/zthings/ht/test/ExampleGame.java (hunk 3 above)
Comment (CruzBishop) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: It's only fixed if you don't resize the window 😢

File: src/main/java/info/zthings/ht/test/RenderContextProvider.java (hunk 6 above)
Comment (CruzBishop) [inline src/main/java/info/zthings/ht/test/RenderContextProvider.java]: um

Comment (CruzBishop) [inline src/main/java/info/zthings/ht/test/RenderContextProvider.java]: Ah shit. Yeah, I missed that :p Sorry, mate

File: src/main/java/info/zthings/ht/test/ExampleGame.java [hunk 7]
```diff
@@ -19,52 +19,64 @@
...
-							jobj2.remove("y2");
-							jobj2.put("x", x);
-							jobj2.put("y", y);
-							
-							map[y][x] = c.newInstance(jobj);
+				Constructor<? extends Tile> constructor = tiletypes.get(key).getConstructor(JSONObject.class);
+				for (Object obj : json.getJSONArray(key)) {
+					JSONObject objectOne = (JSONObject) obj;
+					if (TileField.class.isAssignableFrom(tiletypes.get(key))) {
+						boolean numberedX = objectOne.keySet().contains("x1"); //TODO Maybe find the class and load it per-class instead of this
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: Ok so now that I properly looked into your code (via a primitive - 2 notepadqq windows side-by-side method), I'm 99% sure that I'm going crazy...

So at line 50 I check if the current "tile object json class thingy" (I really hope you can bear with me here...) is an implementor of `TileField` (tiles that fill an area [and thus have x1, y1, etc. coordinates]), this is to ensure that if I ever need to implement different behavior for those types I can do that form the interface.

On line 51 you are checking if the current key contains an x1 value, but that HAS to be the case, or else the key is invalid. (So you could call this a redundant check.) Now what I don't get is that **with** this check it suddenly works :confused: :question:

File: src/main/java/info/zthings/ht/test/ExampleGame.java (hunk 3 above)
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: I'll fix that sometime...somehow...once....ever....

File: src/main/java/info/zthings/ht/test/RenderContextProvider.java (hunk 6 above)
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/RenderContextProvider.java]: No problem, it just stood there pretty weird...

File: src/main/java/info/zthings/ht/test/ExampleGame.java (hunk 2 above)
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: Appearently it didn't work, this code was just never reached :wink:

File: src/main/java/info/zthings/ht/test/ExampleGame.java [hunk 8]
```diff
@@ -19,52 +19,64 @@
...
+						System.out.println("Numbered coordinate setting is " + numberedX);
+
+						if (numberedX) {
+							System.out.println("x1 = " + objectOne.getInt("x1"));
+							System.out.println("y1 = " + objectOne.getInt("y1"));
+							System.out.println("x2 = " + objectOne.getInt("x2"));
+							System.out.println("y2 = " + objectOne.getInt("y2"));
+							for (int yCoordinate = objectOne.getInt("y1"); yCoordinate <= objectOne.getInt("y2"); yCoordinate++) {
+								for (int xCoordinate = objectOne.getInt("x1"); xCoordinate <= objectOne.getInt("x2"); xCoordinate++) {
+									JSONObject objectToInsert = new JSONObject();//objectOne;
```
Comment (HoldYourWaffle) [inline src/main/java/info/zthings/ht/test/ExampleGame.java]: Ok turns out that this line was the **original** culprit. Appearently Maps are passed by reference in java?
Anyhow I'll fix this in a future commit, because what this does is it creates a new object and then puts in X & Y, losing any future properties I want to put in there (high chance I will)
